fdb-record-layer-core/src/test/java/com/apple/foundationdb/record/provider/foundationdb/FDBRecordStoreIndexTest.java

```diff
@@ -1991,6 +1996,11 @@ public void testChangeIndexDefinitionWriteOnly() throws Exception {
             recordMetaData = recordStore.getRecordMetaData();
             // The changed index is again write-only because there are 300 records.
             assertEquals(IndexState.WRITE_ONLY, recordStore.getRecordStoreState().getState(index.getName()));
+            assertEquals(1L, recordStore.getRecordMetaData().getIndex(index.getName()).getSubspaceKey());
+            if (changeIndexNameToo) {
+                MetaDataException ex = assertThrows(MetaDataException.class, () -> recordMetaData.getIndex("index"));
+                assertEquals(ex.getMessage(), "Index index not defined");
+            }
```

Comment:
I don't think this is necessarily a bad test, but the bug I was alluding to was something like: an index exists and is `DISABLED` (or `WRITE_ONLY`, but `DISABLED` for the sake of argument). Then let's say *only* it's name changes. (For the sake of argument, let's also say that it's last modified version doesn't change, which I think the platonic ideal Record Layer wouldn't require it does change.) Then, if the store state caching feature is enabled, it might appear to the user like it went from `DISABLED` to `READABLE` because the state is cached by the old name.

Just clarifying here in case this test was designed to expose the bug I was alluding to.

Reply:
Sure, added one.